### ISSUE TITLE
LiteStar Would be better fit because it have full HTMX support natively.

### ISSUE THREAD

**Issue by v3ss0n** [paveldedik/ludic#65]
Have you tried Litestar?  It is faster than starlette and it have native HTMX support  along with many other features.

**Comment by paveldedik** [comment]
Thank you for the issue @v3ss0n. I have heard about Litestar but haven't checked it that much yet, only glimpsed I'd say.

I think it would be nice to add optional support (even Starlette is optional support). I didn't know it has a native HTMX support, I'll definitely check it out.

**Comment by rupurt** [comment]
+1 would love to use this with Litestar

**Comment by jkatzer** [comment]
@paveldedik I think this is a good reason for ludic to:
1. be framework agnostic
1. just handle the concern of type-safe html
1. additionally support htmx in a modular way (but still part of ludic) to show how people can use ludic's type-safe html generation guarantees for any situation of generating html. if it were modular, it could do things like code-generator react components, or even swiftui components.

I think it makes sense for ludic to focus on type-safe generation of html (and possibly other text based GUI structures). it should be easy to support any framework and possibly for people to bring their own.

I think a great example is "[Bolt](https://slack.dev/bolt-python/concepts)" from Slack. ([repo](https://github.com/SlackAPI/bolt-python)).

It really nails the use case of providing simple abstractions, reducing work for developers and supporting a list of different web frameworks. (It can also run on its own, which I thought you might like). But... it missed the boat on modularity. They could have made it easy to support other chat systems.

**Comment by paveldedik** [comment]
@jkatzer Yeah, I agree with you. I think it will be my main focus in the following months - improving type-safe html. Integration with other frameworks is lower priority (and somewhat up to the community, if they want it faster). My main focus is building reliable type-safe HTML generator, basically.

Thank you for the link to Bolt, I need to do some research when it comes to modularity, etc.

**Comment by mecampbellsoup** [comment]
Is it not possible to just use Ludic OOTB w/ LiteStar? We're trying to decide between LiteStar and Ludic+Django (or some other option) for a new project.

**Comment by paveldedik** [comment]
Hi @mecampbellsoup it should definitely be possible to use with litestar, you can check how ludic.contrib.django is implemented (it basically only adds a middleware to cleanup possible memory leaks after improperly using f-strings with ludic) but even without any additional middleware, you could still use ludic to generate HTML. It is also possible to integrate Ludic somehow that you can directly return ludic components from views (which is how the fastapi integration works). Not sure how that could be done in LiteStar, but it is possible that just a simple middleware calling the .to_html() method and wrapping it in a LiteStar response object would suffice - which is something you would otherwise have to do in your views (or you could write an abstract view that would do that for you).
